rclcpp/src/rclcpp/subscription_base.cpp

```diff
@@ -354,3 +356,89 @@ SubscriptionBase::set_on_new_message_callback(
     throw_from_rcl_error(ret, "failed to set the on new message callback for subscription");
   }
 }
+
+#ifdef CONTENT_FILTER_ENABLE
+bool
+SubscriptionBase::is_cft_enabled() const
+{
+  return rcl_subscription_is_cft_enabled(subscription_handle_.get());
+}
+
+void
+SubscriptionBase::set_content_filter(
+  const std::string & filter_expression,
+  const std::vector<std::string> & expression_parameters)
+{
+  rcl_subscription_content_filter_options_t options =
+    rcl_get_zero_initialized_subscription_content_filter_options();
+
+  std::vector<const char *> cstrings =
+    get_c_vector_string(expression_parameters);
+  rcl_ret_t ret = rcl_subscription_content_filter_options_init(
+    get_c_string(filter_expression),
+    cstrings.size(),
+    cstrings.data(),
+    &options);
```

Comment:
I forgot, does `rcl_subscription_content_filter_options_init()` make a copy of the expression parameters vector internally?
i.e. does it copy `cstrings` instead of taking ownership of the passed pointer?

In retrospective, maybe taking ownership of the passed pointer was a better idea, we would've needed to make less copy here.
Is the behavior clear in `rmw` documentation?

Reply:
Yes, it uses the deep copy in `rmw_subscription_content_filter_options_t` (`rcl_subscription_content_filter_options_init` calls rmw_subscription_content_filter_options_init`).
https://github.com/ros2/rmw/blob/d778df84af42cdbc5458361a2fe220317494634c/rmw/src/subscription_content_filter_options.c#L69 . 
If it just takes the ownership of the pointer, I think it should be documented.